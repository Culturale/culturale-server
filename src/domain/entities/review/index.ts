export * from './review';
export * from './review.interface';
export * from './review.schema';
