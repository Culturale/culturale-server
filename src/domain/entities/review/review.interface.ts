import type { MongoId } from '~/types/types';

export interface IReview {
  _id: MongoId;
  author: string;
  eventId: string; 
  puntuation: number;
  comment?: string;     
}
