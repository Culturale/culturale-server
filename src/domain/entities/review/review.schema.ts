import { Schema, model } from 'mongoose';

import type { IReview } from './review.interface';

const ReviewSchema = new Schema({
  authorId: { required: true, type: String },
  eventId: { required: true, type: String }, 
  puntuation: { required: true, type: Number},
  comment: { required: false, type:  String},
  report: { required: true, type: Number, default: 0 }
});

const ReviewModel = model<IReview>('Review', ReviewSchema);

export { ReviewModel };
