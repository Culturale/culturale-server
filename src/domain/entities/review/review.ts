import type { MongoId } from '~/types/types';

import type { IReview } from './review.interface';

export type reviewProps = {
  id?: MongoId;
  puntuation: number;
  comment?: string;
  author: string;
  eventId: string;
};
export class Review implements IReview  {
  public _id: MongoId;
  public puntuation: number;
  public comment?: string;
  public author: string;
  public eventId: string;

  constructor(props: reviewProps) {
    const {id, puntuation, comment, author, eventId} = props;
    this._id = id;
    this.puntuation = puntuation;
    this.comment = comment || null;
    this.author = author;
    this.eventId = eventId;
  }
  
}
