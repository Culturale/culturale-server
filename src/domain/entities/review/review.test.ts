
import type { MakeReviewDTO } from '~/infrastructure/dtos/make-review.dto';

import { Review } from './review';


describe('Event Review', function () {
  let instance: Review;

  const dateSpy = jest.spyOn(Date, 'now').mockImplementation(() => 1234);

  beforeEach(function () {
    const reviewProps: MakeReviewDTO  = {eventId: '645b9ce003aaad67d81d2887', authorId: '645b9cf103aaad67d81d288a', puntuation: 7, comment: 'comment-test', report: 0};
    instance = new Review(
      reviewProps,
    );
  });

  it('can be created', function () {
    expect(instance).toMatchSnapshot();
    dateSpy.mockReset();
  });
});
