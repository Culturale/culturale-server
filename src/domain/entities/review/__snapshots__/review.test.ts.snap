// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Review can be created 1`] = `
Review {
  "_id": undefined,
  "authorId": "645b9cf103aaad67d81d288a",
  "comment": "comment-test",
  "eventId": "645b9ce003aaad67d81d2887",
  "puntuation": 7,
  "report": 0,
}
`;
